### ISSUE TITLE
Why is `AtomicAtomicRef` always disabled?

### ISSUE THREAD

**Issue by chillenzer** [alpaka-group/alpaka#2287]
Hi,
quick question: What's the reason for [this line](https://github.com/alpaka-group/alpaka/blame/c2a37cc8c9dd82b45a203318bf6000febc460f45/include/alpaka/atomic/AtomicCpu.hpp#L10)? Explicitly:
```C++
#ifndef ALPAKA_DISABLE_ATOMIC_ATOMICREF
#    define ALPAKA_DISABLE_ATOMIC_ATOMICREF
#endif
```
This ALWAYS AND UNCONDITIONALLY disables the `AtomicAtomicRef` path.

**Comment by fwyzard** [comment]
Ops...

It used to be wrong in a different way:
```c++
#ifndef ALPAKA_DISABLE_ATOMIC_ATOMICREF
#    if BOOST_VERSION < 107300
#        warning boost::atomic_ref requires Boost version 1.73 or higher. Please update your version of Boost, or disable the use of boost::atomic_ref with -DALPAKA_DISABLE_ATOMIC_ATOMICREF
#    endif
#    define ALPAKA_DISABLE_ATOMIC_ATOMICREF
#endif
```

Then, when we removed the Boost version check, we left the wrong definition.

**Comment by fwyzard** [comment]
https://github.com/alpaka-group/alpaka/pull/2288 should fix it.

**Comment by psychocoderHPC** [comment]
fixed with #2288 and #2294
